From 051d3f4e5efe67df3b5c4e56a05096cc6a1c22b3 Mon Sep 17 00:00:00 2001
From: lvyuemeng <lv15623448685@gmail.com>
Date: Sat, 29 Mar 2025 22:27:22 +0800
Subject: [PATCH 1/2] refac(ch3): try...

---
 os/src/syscall/process.rs | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/os/src/syscall/process.rs b/os/src/syscall/process.rs
index 0784fc2..52d5dc7 100644
--- a/os/src/syscall/process.rs
+++ b/os/src/syscall/process.rs
@@ -1,6 +1,9 @@
 //! Process management syscalls
 use crate::{
-    task::{cur_task_syscall, exit_current_and_run_next, suspend_current_and_run_next},
+    task::{
+        cur_task_syscall, cur_task_syscall_record, exit_current_and_run_next,
+        suspend_current_and_run_next,
+    },
     timer::get_time_us,
 };
 
@@ -53,7 +56,11 @@ pub fn sys_trace(_trace_request: usize, _id: usize, _data: usize) -> isize {
             unsafe { addr.write_volatile(byte) };
             0
         }
-        2 => cur_task_syscall(_id) as isize,
+        2 => {
+            let target_id = _id;
+            cur_task_syscall_record(target_id);
+            cur_task_syscall(_id) as isize
+        }
         _ => -1,
     }
 }
-- 
2.43.0

